rasa/shared/nlu/training_data/message.py

```diff
@@ -116,6 +121,10 @@ def build(
                 data[INTENT_RESPONSE_KEY] = intent
         if entities:
             data[ENTITIES] = entities
+        if intent_metadata is not None:
+            data[METADATA] = {METADATA_INTENT: intent_metadata}
+        if metadata is not None:
+            data.setdefault(METADATA, {})[METADATA_EXAMPLE] = metadata
```

Comment:
this will likely need to be documented. Where is the right place for this?

Reply:
I think we need to mention NLU `metadata` in this [doc](https://rasa.com/docs/rasa/next/training-data-format).
@akelad should we mention somewhere that `Message` class now has this metadata parsed?

Reply:
agreed to mentioning it in the training data doc, perhaps we also add a quick explanation of how it can be used for custom components [here](https://rasa.com/docs/rasa/next/components/custom-nlu-components)? Or just link to the custom nlu component page from the training data format in that section.
About the `Message` class, I don't think we have that documented anywhere - but @m-vdb it's worth doing a quick search of the docs files locally to double check

Reply:
@akelad WDYT d1fa79e ? Any suggestions?

Reply:
looks good to me